src/lib/builtins/functions.rs

```diff
@@ -5,15 +5,13 @@ use std::io::{self, Write};
 use types::Identifier;
 
 fn print_functions(functions: &FnvHashMap<Identifier, Function>) {
-    let stdout = io::stdout();
-    let stdout = &mut stdout.lock();
-    let _ = writeln!(stdout, "# Functions");
+    println!("# Functions");
     for fn_name in functions.keys() {
         let description = &functions.get(fn_name).unwrap().get_description();
         if let Some(ref description) = *description {
-            let _ = writeln!(stdout, "    {} -- {}", fn_name, description);
+            println!("    {} -- {}", fn_name, description);
         } else {
-            let _ = writeln!(stdout, "    {}", fn_name);
```

Comment:
This is actually done for performance reasons. Each invocation of `println!()` is costly because it locks, writes, and unlocks. Doing it this way only needs to lock once.

Reply:
jackpot51 suggested doing this. Look at the developer chat. The performance impact is tiny.